kani-driver/src/cbmc_output_parser.rs

```diff
@@ -0,0 +1,989 @@
+        };
+        // TODO: Record processed items and dump them into a JSON file
+        // <https://github.com/model-checking/kani/issues/942>
+    }
+    if result { VerificationStatus::Success } else { VerificationStatus::Failure }
+}
+
+/// Returns an optional formatted item based on the output format
+fn format_item(item: &ParserItem, output_format: &OutputFormat) -> Option<String> {
+    match output_format {
+        OutputFormat::Old => todo!(),
+        OutputFormat::Regular => format_item_regular(item),
+        OutputFormat::Terse => format_item_terse(item),
+    }
+}
+
+/// Formats an item using the regular output format
+fn format_item_regular(item: &ParserItem) -> Option<String> {
+    match item {
+        ParserItem::Program { program } => Some(program.to_string()),
+        ParserItem::Message { message_text, .. } => Some(message_text.to_string()),
+        ParserItem::Result { result } => Some(format_result(result, true)),
+        _ => None,
+    }
+}
+
+/// Formats an item using the terse output format
+fn format_item_terse(item: &ParserItem) -> Option<String> {
+    match item {
+        ParserItem::Result { result } => Some(format_result(result, false)),
+        _ => None,
+    }
+}
+
+/// Formats a result item (i.e., the complete set of verification checks).
+/// This could be split into two functions for clarity, but at the moment
+/// it uses the flag `show_checks` which depends on the output format.
+fn format_result(properties: &Vec<Property>, show_checks: bool) -> String {
+    let mut result_str = String::new();
+    let mut number_tests_failed = 0;
+    let mut number_tests_unreachable = 0;
+    let mut number_tests_undetermined = 0;
+    let mut failed_tests: Vec<&Property> = vec![];
+
+    let mut index = 1;
+
+    if show_checks {
+        result_str.push_str("\nRESULTS:\n");
+    }
+
+    for prop in properties {
+        let name = &prop.property;
+        let status = &prop.status;
+        let description = &prop.description;
+        let location = &prop.source_location;
+
+        match status {
+            CheckStatus::Failure => {
+                number_tests_failed += 1;
+                failed_tests.push(prop);
+            }
+            CheckStatus::Undetermined => {
+                number_tests_undetermined += 1;
+            }
+            CheckStatus::Unreachable => {
+                number_tests_unreachable += 1;
+            }
+            _ => (),
+        }
+
+        if show_checks {
+            // TODO: Add color to status if printing to terminal.
+            // <https://github.com/model-checking/kani/issues/1431>
+            let check_id = format!("Check {}: {}\n", index, name);
+            let status_msg = format!("\t - Status: {}\n", status);
+            let description_msg = format!("\t - Description: \"{}\"\n", description);
+
+            result_str.push_str(&check_id);
+            result_str.push_str(&status_msg);
+            result_str.push_str(&description_msg);
```

Comment:
I think you can use `write!` on a `String`, but I'd not worry about this now, do a follow-up refactoring!